source/hpack.c

```diff
@@ -1459,7 +1461,22 @@ int aws_hpack_encode_header_block(
     for (size_t i = 0; i < num_headers; ++i) {
         struct aws_http_header header;
         aws_http_headers_get_index(headers, i, &header);
-        if (s_encode_header_field(context, &header, output)) {
+        if (header.compression == AWS_HTTP_HEADER_COMPRESSION_USE_CACHE &&
```

Comment:
since splitting the COOKIE header is part of the HTTP/2 spec, and not the HPACK spec, this should be in h2_encoder.c

Or we can just remove it for now since the HTTP/2 spec says we MAY split cookies, but it's not required
https://httpwg.org/specs/rfc7540.html#CompressCookie